jdk398/Shadow_of_Knight_Solution.cpp

```diff
@@ -0,0 +1,167 @@
+/* 
+*********************
+Shadows of the Knight
+*********************
+
+This solves the `shadows of the knight <https://www.codingame.com/ide/puzzle shadows-of-the-knight-episode-1>`_ codeingame problem. 
+
+Problem Solution
+================
+.. image:: Shadow_of_knight.png
+    :width: 75%
+
+
+This problem requires the user to code a process to get batman to a bomb in a building. The building is represented as a 2D array, with position (0,0) being in the top left, and position (w,h) being in the bottom right. In the example picture above batman is in the bottom left quadrant of the building and the bomb is the red circle in the top right of the bulding.
+
+The output that we give the game is the next location that we would like to move. For
+example, in the picture above the bomb is far up and to the right, so the next location 
+we would want to choose would be to a location like (10,2). 
+
+Something that also should be noted about the above picture are the two red lines flanking Batman. These are a graphical representation of what I call "limits". Limits are essentially coordinates, either x or y, that Batman is restricted from going past. So in a future move Batman will use the limits defined before as an outer bounds to help calculate his next move. We decide limits based on the direction that the game gives us for the bomb. For our example above, because the bomb is above and to the right, we know for sure that it is not anywhere below or to the left of our current position. by using this method repeatedly we cut down the search area until we can make our final move with certainty. 
+
+For this solution I use the find function of the string class that the game requires us to import. we will be checking this against the class defined value npos. 
+
+ .. code::
```

Comment:
Use the following syntax so that it's clear this code isn't part of your source, but is a comment.

``` c++
// .. code-block:: c++
//  :linenos:
//
//  if (bomb_dir.find("R") != string::npos)
```